provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/client/RestVlerClient.java

```diff
@@ -0,0 +1,171 @@
+  String truststorePassword;
+
+  @Value("${vler.truststore.location}")
+  String truststoreLocation;
+
+  @Value("${vler.url}")
+  String vlerUrl;
+
+  public RestVlerClient(
+      @Value("${vler.url}") String baseUrl, @Autowired RestTemplate restTemplate) {
+    this.baseUrl = baseUrl.endsWith("/") ? baseUrl : baseUrl + "/";
+    this.restTemplate = restTemplate;
+  }
+
+  /*  Implement this if the above isnt secure enough.
+  @SneakyThrows
+  private RestTemplate secureRestTemplate() {
+      try (InputStream truststoreInputStream = new FileInputStream(new File(vlerTruststore))) {
+          KeyStore ts = KeyStore.getInstance("JKS");
+          ts.load(truststoreInputStream, truststorePassword.toCharArray());
+          SSLContext sslContext = null;
+          TrustManagerFactory trustManagerFactory =
+                  TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+          trustManagerFactory.init(ts);
+          sslContext = SSLContext.getInstance("TLSv1.1");
+          sslContext.init(null, trustManagerFactory.getTrustManagers(), new SecureRandom());
+          HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
+          SSLConnectionSocketFactory socketFactory =
+                  new SSLConnectionSocketFactory(
+                          new SSLContextBuilder()
+                                  .loadTrustMaterial(null, new TrustSelfSignedStrategy())
+                                  .loadKeyMaterial(ts, truststorePassword.toCharArray())
+                                  .build(),
+                          new String[]{"TLSv1.1"},
+                          null,
+                          SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+          HttpClient httpsClient = HttpClients.custom().setSSLSocketFactory(socketFactory).build();
+          return new RestTemplate(new HttpComponentsClientHttpRequestFactory(httpsClient));
+      }
+  }*/
+
+  @SneakyThrows
+  private static <T extends VlerResponse> T handleVlerExceptions(
+      String message, Callable<T> callable) {
+    T response;
+    try {
+      response = callable.call();
+    } catch (HttpClientErrorException.NotFound e) {
+      throw new NotFound(message, e);
+    } catch (HttpClientErrorException.BadRequest e) {
+      throw new BadRequest(message, e);
+    } catch (HttpStatusCodeException e) {
+      throw new SearchFailed(message, e);
+    } catch (Exception e) {
+      throw new VlerException(message, e);
+    }
+    if (response == null) {
+      throw new VlerException(message + ", no VLER response");
+    }
+    if (response.error() != null && isNotBlank(response.error().message())) {
+      throw new VlerException(message + ", " + response.error().message());
+    }
+    return response;
+  }
+
+  @SneakyThrows
+  private String authHeader(String baseUrl, String url, String dateString) {
+    String endpoint = "/" + url.substring(url.indexOf(baseUrl) + baseUrl.length());
```

Comment:
No protection around indexing and substring. This method _assumes_ `url` contains `baseUrl`, but doesn't verify the assumption.